src/gtk/window.cpp

```diff
@@ -3692,10 +3692,18 @@ void wxWindowGTK::ConnectWidget( GtkWidget *widget )
         g_source_unref(source);
     }
 
-    g_signal_connect (widget, "key_press_event",
+    // We need to connect to the key events for the widget which is actually
+    // going to get them, which might be different from the main widget itself,
+    // but in this case the derived class is supposed to set m_focusWidget
+    // before calling PostCreation() (from where we're usually called).
+    GtkWidget* const focusWidget = m_focusWidget && m_focusWidget != m_widget
```

Comment:
It's not clear to me why `&& m_focusWidget != m_widget` is needed.

Reply:
When `ConnectWidget()` is called with a different widget (as e.g. from `src/gtk/radiobox.cpp`) we don't want to override it with `m_focusWidget`. Maybe it would be more clear to write
```cpp
if ( widget == m_widget && m_focusWidget )
  ... use m_focusWidgets ...
```
?

Reply:
Yes, that seems clearer to me.